flink-libraries/flink-table/src/main/scala/org/apache/flink/table/functions/utils/UserDefinedFunctionUtils.scala

```diff
@@ -140,6 +138,24 @@ object UserDefinedFunctionUtils {
           s"one method named 'eval' which is public, not abstract and " +
           s"(in case of table functions) not static.")
     } else {
+      var trailingSeq = false
+      var noVargs = true
+      methods.foreach(method => {
+        val signatures = method.getParameterTypes
+        if (signatures.nonEmpty) {
+          if (method.isVarArgs) {
+            noVargs = false
+          } else if (signatures.last.getName.equals("scala.collection.Seq")) {
+            trailingSeq = true
+          }
+        }
+      })
+      if (trailingSeq && noVargs) {
+        // We found trailing "scala.collection.Seq", but no trailing "Type[]", "Type..."
+        throw new ValidationException("The 'eval' method do not support Scala type of " +
```

Comment:
If there is multiple eval methods found (not ambiguous), one is varargs, the other is not. It seems that no exception is thrown to tell users that the non-varargs eval method is not work.

```
@varargs
def eval(args: String*): String = {...}
// no varargs annotation
def eval(args: Int*): Int = {...}
```